### ISSUE TITLE
BUG: reading using Arrow interface fails for files with only geometries

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/pyogrio#227]
Reproducer:

```
import geopandas
import pyogrio

gdf = geopandas.GeoDataFrame(geometry=geopandas.points_from_xy([0, 0], [1, 1]))
pyogrio.write_dataframe(gdf, "test_only_geometry.gpkg")

# works fine
pyogrio.read_dataframe("test_only_geometry.gpkg")

# fails
pyogrio.read_dataframe("test_only_geometry.gpkg", use_arrow=True)
```

giving 
```
GEOSException: ParseException: Unexpected EOF parsing WKB
```
In practice we see that the WKB values we receive are all empty bytes.

Using pyogrio 0.5.1 and GDAL 3.6.2

---

We noticed this while giving https://github.com/martinfleis/efficient-geopandas-workshop. So @martinfleis in the end it has nothing to do with using a Mac M1 or M2, it was because of our error in the notebook reusing the `gdf` variable from the exercise which was a geodataframe with only one column of linestrings.

**Comment by martinfleis** [comment]
Good debugging!

**Comment by jorisvandenbossche** [comment]
Testing with some other file formats, I don't see this with any other, but only with GeoPackage, so it might be a bug in GDAL (they have a custom implementation for GeoPackage for the `GetArrowStream`)

**Comment by jorisvandenbossche** [comment]
Whoops, totally forgot that we already debugged this once before ;) See https://github.com/geopandas/pyogrio/issues/212, and this is fixed in GDAL, and will be in the upcoming 3.6.3
